test/python/circuit/test_uc.py

```diff
@@ -100,6 +107,21 @@ def test_inverse_ucg(self):
 
         self.assertTrue(np.allclose(unitary_desired, unitary))
 
+    def test_ucge(self):
+        """test ucg simplification"""
+        gate_list = [_had, _had, _had, _had, _had, _had, _had, _had]
+
+        qc1 = QuantumCircuit(4)
+        uc1 = UCGate(gate_list, up_to_diagonal=False, mux_simp=False)
+        qc1.append(uc1, range(4))
+        op1 = Operator(qc1).data
+
+        qc2 = QuantumCircuit(4)
+        uc2 = UCGate(gate_list, up_to_diagonal=False, mux_simp=True)
+        qc2.append(uc2, range(4))
+        op2 = Operator(qc2).data
+        self.assertTrue(np.allclose(op1, op2))
```

Comment:
When I change `up_to_diagonal=True` in both lines 115 and 120, then the test fails, any idea why? 
If `mux_simp` should not work in case that `up_to_diagonal=True` - then it's better to state it in the docs and raise a proper error that asserts that the the two variables can not both be `True`.

Reply:
This occurs because the final operator when we have `up_to_diagonal=True` is not equal for the original and the simplified multiplexers. But when we multiply the two operators by the respective diagonal matrices obtained by `self._get_diagonal()`, the final operators (and the prepared states) are the same.